BUG: invalid cm/tm in visitor functions (#2206)
py-pdf/pypdf: docs/user/extract-text.md

You can use visitor-functions to control which part of a page you want to process and extract. The visitor-functions you provide will get called for each operator or for each text fragment.

The function provided in argument visitor_text of function extract_text has five arguments:

> * text : the current text (as long as possible, can be up to a full line)

Reviewer: ```suggestion
* text: the current text (as long as possible, can be up to a full line)
```

> * user_matrix: current matrix to move from user coordinate space(aka. CTM)

Reviewer: ```suggestion
* user_matrix: current matrix to move from user coordinate space (also known as CTM)
```

* tm_matrix: current matrix from text coordinate space
* font-dictionary: full font dictionary
* font-size: the size (in text coordinate space)

> the matrix stores 6 parameters. the 4 first provides the rotation/scaling matrix and the last two provides the translation (horizontal/vertical)

Reviewer: ```suggestion
The matrix stores 6 parameters. The first 4 provide the rotation/scaling matrix and the last two provide the translation (horizontal/vertical)
```

> it is recommended to use the user_matrix as it takes into all transformations.

Reviewer: ```suggestion
It is recommended to use the user_matrix as it takes into all transformations.
```

> notes :
>  - as indicated in pdf 1.7 refeence, page 204 the user matrix applies to text space/image space/form space/pattern space.
>  - if you want to get the full transformation from text to user space, you can use the mult function (availalbe in global import) as follow :
> `txt2user = mult(tm, cm))`
> the font-size is the raw text size, that is affected by the user_matrix

Reviewer: ```suggestion
Notes :

 - as indicated in the PDF 1.7 reference, page 204 the user matrix applies to text space/image space/form space/pattern space.
 - if you want to get the full transformation from text to user space, you can use the `mult` function (availalbe in global import) as follows:
`txt2user = mult(tm, cm))`
The font-size is the raw text size, that is affected by the `user_matrix`
```
